arkouda/dataframe.py

```diff
@@ -1434,11 +1475,13 @@ def copy(self, deep=True):
             res._size = self._size
             res._bytes = self._bytes
             res._empty = self._empty
-            res._columns = self._columns
+            res._columns = self._columns[:]  # if this is not a slice, droping columns modifies both
 
             for key, val in self.items():
                 res[key] = val[:]
 
+            res._set_index(Index(self.index.index))
```

Comment:
Just double checking this is supposed to be `self.index.index` and not `self.index`?